### PR TITLE
CLDR-19466 Document decimal format test data files and usage in README

### PR DESCRIPTION
**Pull request by younies**
CLDR-19466

- [ ] This PR completes the ticket.

### Summary
Addressed PR #5709 feedback by adding a comprehensive `README.md` to `common/testData/decimal/`.

### Key Details
* **File Scopes**: Documented what is in each test file, explicitly clarifying that `decimals_modern_locales.tsv` covers modern locales **minus** the core locales in `decimals.tsv`.
* **TSV Schema**: Defined all columns (`locale`, `number_format`, `format_length`, `input`, `expected`).
* **Testing & Maintenance**: Added instructions for automated test execution (`TestDecimalFormat`), manual spreadsheet inspection, and data regeneration (`GenerateDecimalFormatTestData`).

ALLOW_MANY_COMMITS=true

### CODE REVIEW

File: common/testData/decimal/README.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
+# CLDR Decimal Format Test Data (`common/testData/decimal`)
+
+This directory contains Tab-Separated Values (TSV) files used as a solid blueprint for testing and verifying decimal and compact decimal formats in CLDR, based on ICU4J.
```
Comment (macchiati) [inline common/testData/decimal/README.md]:
```suggestion
This directory contains Tab-Separated Values (TSV) files used for testing decimal and compact decimal formats in CLDR.
```
This is more accurate and avoids the awkward phrasing "solid blueprint"

File: common/testData/decimal/README.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+## Test Data Files
+
+The test data is organized into three separate files to separate core verification from extended coverage:
+
+1. **`decimals.tsv`**
+   Contains verification tests for the **core** CLDR locales across standard number formats, compact format lengths, and representative numeric input values.
```
Comment (macchiati) [inline common/testData/decimal/README.md]:
```suggestion
   Contains verification tests for a selected set of numbers and locales that illustrate most features of number formatting, covering standard number formats, compact format lengths, and representative numeric input values.
```

File: common/testData/decimal/README.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+2. **`decimals_modern_locales.tsv`**
+   Contains verification tests for all **modern** CLDR locales **minus** the core locales covered in `decimals.tsv`.
```
Comment (macchiati) [inline common/testData/decimal/README.md]:
```suggestion
   Contains verification tests for all **modern-coverage** CLDR locales **minus** the locales covered in `decimals.tsv`. It also only uses a small set of selected numbers.
```

File: common/testData/decimal/README.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+3. **`decimals_extended_numbers.tsv`**
+   Contains extended numeric test inputs (covering edge cases, large numbers, and small fractions) across standard formats and core locales for comprehensive verification.
```
Comment (macchiati) [inline common/testData/decimal/README.md]:
```suggestion
   Contains extended numeric test inputs (covering edge cases, large numbers, and small fractions) across standard formats and selected locales for more comprehensive verification.
```

File: common/testData/decimal/README.md [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+```tsv
+locale	number_format	format_length	input	expected
+```
+
+### Column Definitions:
+* **`locale`**: The CLDR locale identifier (e.g., `ar`, `en`, `es_AR`).
+* **`number_format`**: The number format dimension (`decimal`, `percent`, `scientific`).
+* **`format_length`**: The compact format length (`short`, `long`). If empty/blank, it represents standard non-compact formatting.
+* **`input`**: The floating-point numeric input value (e.g., `1.2`, `1234565.0`).
+* **`expected`**: The literal expected output string, including all correct localized digits, grouping separators, percent signs, and bi-directional control characters (such as `\u200E` / LRM).
```
Comment (macchiati) [inline common/testData/decimal/README.md]:
```suggestion
* **`expected`**: The expected output string, including all correct localized digits, grouping separators, percent signs, and bi-directional control characters (such as `\u200E` / LRM).
```

File: common/testData/decimal/README.md [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+## How to Read and Use the Files
+
+### 1. Automated Testing
+
+#### Testing in Java
```
Comment (macchiati) [inline common/testData/decimal/README.md]: Drop this whole section

File: common/testData/decimal/README.md [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+* Blank lines and lines starting with `#` are ignored.
+* The `input` number is formatted using ICU4J (`GenerateDecimalFormatTestData.format(...)`) according to the specified `locale`, `number_format`, and `format_length`.
+* The actual output is asserted against the literal `expected` string.
+
+To run the Java tests locally from the command line:
+```bash
+mvn test -pl tools/cldr-code -Dtest=TestDecimalFormat
+```
+
+#### Testing in Other Languages
```
Comment (macchiati) [inline common/testData/decimal/README.md]: Drop this whole section also

File: common/testData/decimal/README.md [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+If you want to use this test data with other programming languages or formatting engines (such as ICU4C, ICU4X, or JavaScript/Intl), you will need to implement a TSV reader that:
+1. Skips the header line, blank lines, and comments (lines starting with `#`).
+2. Parses the five tab-separated columns (`locale`, `number_format`, `format_length`, `input`, `expected`).
+3. Invokes your target number formatting implementation using the parsed parameters and numeric input.
+4. Asserts that the generated output exactly matches the `expected` UTF-8 literal string.
+
+### 2. Manual Verification
```
Comment (macchiati) [inline common/testData/decimal/README.md]: Dropped the numbering on these sections

**Review by macchiati — CHANGES_REQUESTED** [review]

**Review by sffc — COMMENTED** [review]
+1 on all of @macchiati's comments

File: common/testData/decimal/README.md (hunk 1 above)
Comment (younies) [reply common/testData/decimal/README.md]: done

File: common/testData/decimal/README.md (hunk 2 above)
Comment (younies) [reply common/testData/decimal/README.md]: done

File: common/testData/decimal/README.md (hunk 3 above)
Comment (younies) [reply common/testData/decimal/README.md]: done

File: common/testData/decimal/README.md (hunk 4 above)
Comment (younies) [reply common/testData/decimal/README.md]: done

File: common/testData/decimal/README.md (hunk 5 above)
Comment (younies) [reply common/testData/decimal/README.md]: done

File: common/testData/decimal/README.md (hunk 6 above)
Comment (younies) [reply common/testData/decimal/README.md]: done

File: common/testData/decimal/README.md (hunk 7 above)
Comment (younies) [reply common/testData/decimal/README.md]: done

File: common/testData/decimal/README.md (hunk 8 above)
Comment (younies) [reply common/testData/decimal/README.md]: done

**Comment by younies** [comment]
thanks @macchiati for the comments, I will follow the same style with the currency testing too.

**Review by sffc — APPROVED** [review]
The prose is fine, though in currency data, where we have more dimensions, I think we should be more clear and specific about which sets of options are used where.

**Review by macchiati — APPROVED** [review]
